contracts/InterchainTokenFactory.sol

```diff
@@ -285,6 +286,8 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
             } else {
                 chainNameHash_ = keccak256(bytes(originalChain));
             }
+            if (chainNameHash_ == keccak256(bytes(destinationChain))) revert CannotDeployRemotely(destinationChain);
```

Comment:
I think we want to remove the ability to deploy a token with a different `originalChain` instead. And perform this check inside ITS `deploy*` methods instead so you can never submit a self deploy call

Reply:
Had offline discussion. updated.